src/Moryx.Communication.Mqtt.Services/ResourceEventService.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG
+// Licensed under the Apache License, Version 2.0
+
+using Moryx.AbstractionLayer.Identity;
+using Moryx.AbstractionLayer.Resources;
+using Moryx.Communication.Mqtt.Builders;
+using Moryx.Communication.Mqtt.Components;
+using Moryx.Runtime.Modules;
+using MQTTnet.Extensions.ManagedClient;
+
+namespace Moryx.Communication.Mqtt.Services;
+
+/// <summary>
+/// Sends messages to the broker when a public event or event exposed using the <see cref="ResourceAvailableAsAttribute"/>
+/// is raised
+/// </summary>
+/// <param name="resourceManagement">Resource management facade</param>
+public class ResourceEventService : IMqttService
+{
+    private readonly IManagedMqttClient _client;
+    private readonly IResourceManagement _resourceManagement;
+    private readonly MqttClientUserOptions _options;
+    public ResourceEventService(
+        IManagedMqttClient client,
+        IResourceManagement resourceManagement,
+        MqttClientUserOptions options)
+    {
+        ArgumentNullException.ThrowIfNull(resourceManagement);
+        ArgumentNullException.ThrowIfNull(client);
+        ArgumentNullException.ThrowIfNull(options);
+        _resourceManagement = resourceManagement;
+        _client = client;
+        _options = options;
+    }
+
+    #region Hosted Service
+    public Task StartAsync(CancellationToken cancellationToken)
+    {
+        if (_resourceManagement is ILifeCycleBoundFacade lifeCycleBoundFacade)
+        {
+            lifeCycleBoundFacade.StateChanged += FacadeStateChanged;
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public Task StopAsync(CancellationToken cancellationToken)
+    {
+        if (_resourceManagement is ILifeCycleBoundFacade lifeCycleBoundFacade)
+        {
+            lifeCycleBoundFacade.StateChanged -= FacadeStateChanged;
+        }
+        return Task.CompletedTask;
+    }
+    #endregion
+
+    private void FacadeStateChanged(object? sender, bool activated)
+    {
+        if (activated)
```

Comment:
As soon as a single restart happens during runtime you will have 2 subscriptions each. You also need to unsubscribe here 🙂  
_Porbably also true for the other services?_